samples/nrf9160/pdn/src/main.c

```diff
@@ -69,6 +80,52 @@ void main(void)
 
 	printk("Default APN is %s\n", apn);
 
+	err = nrf_getifaddrs(&ifaddrs);
+	if (err) {
+		printk("nrf_getifaddrs() failed, err %d\n", err);
+		return;
+	}
+
+	ifaddr = ifaddrs;
+
+	printk("\nInterface addresses:\n");
+
+	while (ifaddr != NULL) {
+		switch (ifaddr->ifa_netmask->sa_family) {
+		case NRF_AF_INET:
+			nrf_inet_ntop(ifaddr->ifa_addr->sa_family,
+				      &SIN(ifaddr->ifa_addr)->sin_addr,
+				      addr, sizeof(addr));
+			break;
+		case NRF_AF_INET6:
+			nrf_inet_ntop(ifaddr->ifa_addr->sa_family,
+				      &SIN6(ifaddr->ifa_addr)->sin6_addr,
+				      addr, sizeof(addr));
+			break;
+		default:
+			printk("Unknown address family, %d\n", ifaddr->ifa_netmask->sa_family);
+		}
+		nrf_inet_ntop(ifaddr->ifa_netmask->sa_family,
+			      &SIN(ifaddr->ifa_netmask)->sin_addr,
+			      netmask, sizeof(netmask));
+		nrf_inet_ntop(ifaddr->ifa_broadaddr->sa_family,
+			      &SIN(ifaddr->ifa_broadaddr)->sin_addr,
+			      broadaddr, sizeof(broadaddr));
+		nrf_inet_ntop(ifaddr->ifa_dstaddr->sa_family,
+			      &SIN(ifaddr->ifa_dstaddr)->sin_addr,
+			      dstaddr, sizeof(dstaddr));
+
+		printk("Address: %s\n", addr);
+		printk("Netmask: %s\n", netmask);
+		printk("Broadcast address: %s\n", broadaddr);
+		printk("Destination address: %s\n", dstaddr);
+		printk("\n");
+
+		ifaddr = ifaddr->ifa_next;
+	}
+
+	nrf_freeifaddrs(ifaddrs);
```

Comment:
I think it would be cleaner to move this `getifaddrs` section to a function just to keep everything tidy.